services/supervisor/src/api.rs

```diff
@@ -94,10 +100,10 @@ impl PublicApi for ApiImpl<'_> {
         Ok(CoreSchema::new(self.0.snapshot()).consensus_config())
     }
 
-    fn config_proposal(&self) -> Result<Option<ConfigProposalWithHash>, Self::Error> {
+    fn config_proposal(&self) -> Result<Vec<ConfigProposalWithHash>, Self::Error> {
```

Comment:
If many proposals will be sent, this method can become very heavy. I would suggest to introduce a constant like [`MAX_BLOCKS_PER_REQUEST`](https://github.com/exonum/exonum/blob/master/exonum/src/api/node/public/explorer.rs#L42), but with smaller value (e.g. 50).